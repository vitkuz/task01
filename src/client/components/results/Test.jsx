import React from 'react';
import Item from './item';


class Test extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            dummy: [],
            value: '',
        };
        this.handleClick = this.handleClick.bind(this);
        this.handleInputChange = this.handleInputChange.bind(this);
    }
    handleClick() {
        const {dummy} = this.state;
        const ids = this.state.dummy.map(item => item.id);
        const maxId = ids.length > 0 ? Math.max(...ids) : 0;
        dummy.push({id: maxId + 1});
        this.setState({dummy});
    }
    handleInputChange(e) {
        this.setState({ value: e.target.value });
    }
    render() {
        return (
            <div>
                <div>
                    <form action="">
                        <input type="text" className="input1" onChange={this.handleInputChange} value={this.state.value} />
                    </form>
                </div>
                <div>
                    <a role="button" tabIndex="-1" className="btn" onClick={this.handleClick}>Click</a>
                </div>
                <div className="item-list">
                    {
                        this.state.dummy.map((item) => {
                            return (
                                <Item key={item.id} />
                            );
                        })
                    }
                </div>
            </div>
        );
    }
}

export default Test;
