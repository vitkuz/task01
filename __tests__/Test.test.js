import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Test from '../src/client/components/results/Test';

configure({ adapter: new Adapter() });

describe('Test', () => {
    const test = shallow(<Test />);
    it('Renders correctly', () => {
        expect(test).toMatchSnapshot();
    });
    it('Test state', () => {
        expect(test.state().dummy).toEqual([]);
    });
    describe('When clicking', () => {
        beforeEach(() => {
            test.find('.btn').simulate('click');
        });
        afterEach(() => {
            test.setState({ dummy: [], value: '' });
        });
        it('Find link, click on link, object added to state', () => {
            expect(test.state().dummy).toEqual([{ id: 1 }]);
        });
        it('Find link, click on link, test object rendered', () => {
            expect(test.find('.item-list').children().length).toEqual(1);
        });
    });
    describe('When changing input update `sate` ', () => {
        const stringToTest = 'Qwerty';
        beforeEach(() => {
            test.find('.input1').simulate('change', { target: { value: stringToTest } });
        });
        it('Find link, click on link, object added to state', () => {
            expect(test.state().value).toEqual(stringToTest);
        });
    });
});

// it('should dispatch action', (done) => {
//     const getState = {}; // initial state of the store
//     const action = { type: 'SET_SEARCH_QUERY' };
//     const expectedActions = [action];
//     const store = mockStore(getState, expectedActions, done);
//     store.dispatch(action);
// })
