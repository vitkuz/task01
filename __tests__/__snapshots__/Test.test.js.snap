// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Test
    qwerty="123456"
    testFn={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div>
          <h1
            className="title"
          >
            123456
          </h1>
          <form
            action=""
          >
            <input
              className="input1"
              onChange={[Function]}
              type="text"
              value=""
            />
          </form>
        </div>,
        <div>
          <a
            className="btn"
            onClick={[Function]}
            role="button"
            tabIndex="-1"
          >
            Click
          </a>
          <a
            className="btn-fn"
            onClick={[Function]}
            role="button"
            tabIndex="-1"
          >
            Click
          </a>
        </div>,
        <div
          className="item-list"
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1
              className="title"
            >
              123456
            </h1>,
            <form
              action=""
            >
              <input
                className="input1"
                onChange={[Function]}
                type="text"
                value=""
              />
            </form>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": "123456",
              "className": "title",
            },
            "ref": null,
            "rendered": "123456",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "action": "",
              "children": <input
                className="input1"
                onChange={[Function]}
                type="text"
                value=""
              />,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "className": "input1",
                "onChange": [Function],
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            "type": "form",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <a
              className="btn"
              onClick={[Function]}
              role="button"
              tabIndex="-1"
            >
              Click
            </a>,
            <a
              className="btn-fn"
              onClick={[Function]}
              role="button"
              tabIndex="-1"
            >
              Click
            </a>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": "Click",
              "className": "btn",
              "onClick": [Function],
              "role": "button",
              "tabIndex": "-1",
            },
            "ref": null,
            "rendered": "Click",
            "type": "a",
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": "Click",
              "className": "btn-fn",
              "onClick": [Function],
              "role": "button",
              "tabIndex": "-1",
            },
            "ref": null,
            "rendered": "Click",
            "type": "a",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": Array [],
          "className": "item-list",
        },
        "ref": null,
        "rendered": Array [],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div>
            <h1
              className="title"
            >
              123456
            </h1>
            <form
              action=""
            >
              <input
                className="input1"
                onChange={[Function]}
                type="text"
                value=""
              />
            </form>
          </div>,
          <div>
            <a
              className="btn"
              onClick={[Function]}
              role="button"
              tabIndex="-1"
            >
              Click
            </a>
            <a
              className="btn-fn"
              onClick={[Function]}
              role="button"
              tabIndex="-1"
            >
              Click
            </a>
          </div>,
          <div
            className="item-list"
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h1
                className="title"
              >
                123456
              </h1>,
              <form
                action=""
              >
                <input
                  className="input1"
                  onChange={[Function]}
                  type="text"
                  value=""
                />
              </form>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": "123456",
                "className": "title",
              },
              "ref": null,
              "rendered": "123456",
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "action": "",
                "children": <input
                  className="input1"
                  onChange={[Function]}
                  type="text"
                  value=""
                />,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": null,
                "nodeType": "host",
                "props": Object {
                  "className": "input1",
                  "onChange": [Function],
                  "type": "text",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              "type": "form",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <a
                className="btn"
                onClick={[Function]}
                role="button"
                tabIndex="-1"
              >
                Click
              </a>,
              <a
                className="btn-fn"
                onClick={[Function]}
                role="button"
                tabIndex="-1"
              >
                Click
              </a>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": "Click",
                "className": "btn",
                "onClick": [Function],
                "role": "button",
                "tabIndex": "-1",
              },
              "ref": null,
              "rendered": "Click",
              "type": "a",
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": "Click",
                "className": "btn-fn",
                "onClick": [Function],
                "role": "button",
                "tabIndex": "-1",
              },
              "ref": null,
              "rendered": "Click",
              "type": "a",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": Array [],
            "className": "item-list",
          },
          "ref": null,
          "rendered": Array [],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
